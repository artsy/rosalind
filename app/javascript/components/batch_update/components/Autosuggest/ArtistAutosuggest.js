import React from 'react'
import PropTypes from 'prop-types'
import GenericAutosuggest from './GenericAutosuggest'
import { matchArtists } from 'lib/rosalind-api'
import { getSuggestionValue, renderSuggestion } from './helpers'

function ArtistAutosuggest(props) {
  return (
    <GenericAutosuggest
      id="artist-autosuggest"
      placeholder={props.placeholder}
      fetchSuggestions={matchArtists}
      getSuggestionValue={getSuggestionValue}
      renderSuggestion={renderSuggestion}
      selectSuggestion={artist => {
        props.onSelectArtist(artist)
      }}
    />
  )
}

ArtistAutosuggest.propTypes = {
  onSelectArtist: PropTypes.func,
}

ArtistAutosuggest.defaultProps = {
  placeholder: 'Select an artist',
}

export { ArtistAutosuggest }
