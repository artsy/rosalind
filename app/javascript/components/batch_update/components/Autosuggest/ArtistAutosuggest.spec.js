import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { ArtistAutosuggest } from './ArtistAutosuggest'

const mockHandler = jest.fn()

it('renders correctly', () => {
  const rendered = renderer.create(
    <ArtistAutosuggest onSelectArtist={mockHandler} />
  )
  const tree = rendered.toJSON()
  expect(tree).toMatchSnapshot()
})
