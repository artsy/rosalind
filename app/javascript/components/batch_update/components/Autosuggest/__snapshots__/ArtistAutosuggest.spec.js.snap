// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="GenericAutosuggest"
>
  <div
    aria-expanded={false}
    aria-haspopup="listbox"
    aria-owns="react-autowhatever-artist-autosuggest"
    className="react-autosuggest__container"
    role="combobox"
  >
    <input
      aria-activedescendant={null}
      aria-autocomplete="list"
      aria-controls="react-autowhatever-artist-autosuggest"
      autoComplete="off"
      className="react-autosuggest__input"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      placeholder="Select an artist"
      type="text"
      value=""
    />
    <div
      className="react-autosuggest__suggestions-container"
      id="react-autowhatever-artist-autosuggest"
      role="listbox"
    />
  </div>
</div>
`;
